freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UmsUsersState.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.freeipa.service.freeipa.user.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class UmsUsersState {
+
+    private final UsersState usersState;
+
+    private final Map<String, WorkloadCredential> usersWorkloadCredentialMap;
+
+    private final Set<FmsUser> requestedWorkloadUsers;
+
+    public UmsUsersState(UsersState usersState, Map<String, WorkloadCredential> usersWorkloadCredentialMap, Set<FmsUser> requestedWorkloadUsers) {
+        this.usersState = requireNonNull(usersState, "UsersState is null");
+        this.usersWorkloadCredentialMap = requireNonNull(usersWorkloadCredentialMap, "workload credential map is null");
+        this.requestedWorkloadUsers = requireNonNull(requestedWorkloadUsers, "requested workload users is null");
+    }
+
+    public UsersState getUsersState() {
+        return usersState;
+    }
+
+    public Map<String, WorkloadCredential> getUsersWorkloadCredentialMap() {
+        return usersWorkloadCredentialMap;
+    }
+
+    public Set<FmsUser> getRequestedWorkloadUsers() {
+        return requestedWorkloadUsers;
+    }
+
+    public static class Builder {
+        private UsersState usersState;
+
+        private Map<String, WorkloadCredential> workloadCredentialMap = new HashMap<>();
+
+        private Set<FmsUser> requestedWorkloadUsers = new HashSet<>();
+
+        public void setUsersState(UsersState usersState) {
+            this.usersState = usersState;
+        }
+
+        public void addWorkloadCredentials(String userName, WorkloadCredential creds) {
+            workloadCredentialMap.put(userName, creds);
+        }
+
+        public void addRequestedWorkloadUsers(FmsUser user) {
+            requestedWorkloadUsers.add(user);
+        }
```

Comment:
builders' methods usually returns with `this` so you can chain methods

Reply:
done